ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StepSchemaResolver.java

```diff
@@ -160,6 +162,22 @@ private LogicalSchema handleStreamFlatMap(
     );
   }
 
+  private LogicalSchema handleGroupBy(
+      final LogicalSchema sourceSchema,
+      final StreamGroupBy<?> streamGroupBy
+  ) {
+    final List<Expression> groupBy = streamGroupBy.getGroupByExpressions();
+
```

Comment:
Can't this just be:
`return GroupByParamsFactory.build(sourceSchema, streamGroupBy.getGroupByExpressions().getSchema();`

Reply:
Nope. We'd need to first compile the expressions into ExpressionMetadata, which seems a little unnecessary / costly.

Though avoiding that could be a case of premature optimisation... thoughts?